torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -271,6 +285,42 @@ static void UpdateScalarTypeForOutput(
       CreateProfiledTensorTypeWithScalarType(output_tensor_type, scalar_type));
 }
 
+static void RecoverScalarTypeForOutput(
+    Node* n,
+    const c10::ScalarType& scalar_type) {
+  const int64_t onnx_type = ScalarTypeToONNXType(scalar_type);
+  Node* cast_node = n->owningGraph()->create(onnx::Cast, 1);
+  cast_node->addInput(n->output());
+  cast_node->i_(attr::to, onnx_type);
+  cast_node->insertAfter(n);
+  n->output()->replaceAllUsesAfterNodeWith(cast_node, cast_node->output());
+}
+
+static void LowPrecisionCastForStandardOps(Node* n, int opset_version) {
+  TORCH_INTERNAL_ASSERT(n->inputs().size() == 2)
+  TORCH_INTERNAL_ASSERT(n->outputs().size() == 1)
+  auto input1_scalar_type =
+      n->inputs()[0]->type()->castRaw<TensorType>()->scalarType().value();
+  auto input2_scalar_type =
+      n->inputs()[1]->type()->castRaw<TensorType>()->scalarType().value();
+  auto output_scalar_type =
+      n->output()->type()->castRaw<TensorType>()->scalarType().value();
+
+  // The LowPrecision problem will be fixed in ONNX opset 14
+  if (opset_version >= 14) {
+    TORCH_INTERNAL_ASSERT(
+        input1_scalar_type == output_scalar_type ||
```

Comment:
why ||?

Reply:
because inpout1 and input2 may be the different type, so one of them should equal to output scalar type.